File: 27.py

def numberOfSubarrays(nums, k):
    count_odd = 0
    prefix = {0: 1}  # prefix sum count map
    res = 0

    for n in nums:
        if n % 2 != 0:
            count_odd += 1
        res += prefix.get(count_odd - k, 0)
        prefix[count_odd] = prefix.get(count_odd, 0) + 1
    return res

# Example
print(numberOfSubarrays([1,1,2,1,1], 3))  # Output: 2

File: 27.js
function numberOfSubarrays(nums, k) {
  let countOdd = 0;
  let prefix = new Map();
  prefix.set(0, 1);
  let res = 0;

  for (let n of nums) {
    if (n % 2 !== 0) countOdd++;
    res += prefix.get(countOdd - k) || 0;
    prefix.set(countOdd, (prefix.get(countOdd) || 0) + 1);
  }

  return res;
}

console.log(numberOfSubarrays([1, 1, 2, 1, 1], 3)); // Output: 2
